src/applications/vaos/appointment-list/components/BackendAppointmentServiceAlert.jsx

```diff
@@ -1,38 +1,74 @@
 import React from 'react';
+import { useLocation } from 'react-router-dom';
 import { useSelector } from 'react-redux';
 import { selectBackendServiceFailuresInfo } from '../redux/selectors';
-import { FETCH_STATUS } from '../../utils/constants';
+import { FETCH_STATUS, ERROR_CODES } from '../../utils/constants';
+
+function displayType(errorCodes, location) {
+  const isPending = location.pathname.endsWith('/pending');
+  const isPast = location.pathname.endsWith('/past');
+  const isUpcoming = location.pathname.endsWith('/');
+
+  if (errorCodes.some(code => code === 10000 || code === 10006) && isPending) {
```

Comment:
Are we supposed to look for specific error codes in the response? I was under the impression that we should display this for any error that comes through? @simiadebowale, can you clarify?

Reply:
> Are we supposed to look for specific error codes in the response? I was under the impression that we should display this for any error that comes through? @simiadebowale, can you clarify?

I thought the generic message was regardless of which system is down: VSP, VVS and/or HSRM; display the message if any one of those systems experience failure retrieving booked appointment, request appointment or both.

Reply:
Yes, I believe that is correct, I was a bit unclear I guess. I meant the specific error codes (10006, 10000, etc). I wasn't sure if those would be entirely consistent or if we even knew all the possible error codes, that is why I originally was just checking for the length of that array. 

If we know all the error codes, then this is great and more precise.

Reply:
I guess that's why we need @simiadebowale get some clarification 😉

Reply:
@ryanshaw - we're not using code/system to deffrentiate the system returning errors but we want to differentiate between errors fetching pending appts and booked appts

Reply:
@simiadebowale Gotcha. That makes sense.

Reply:
I don't know if this matters right now but the error messages are the same for all the error codes.

Reply:
Is the goal to use the error message returned from the api call?

Reply:
@vbahinwillit - we're still displaying the generic error message. There are cases where BE systems aren't down for fetching upcoming/past appts but are down for fetching Pending appts. In this case, we want to only display this generic error message when the user is viewing their pending appts list and not when they're viewing their upcoming/past appts list

Reply:
Still confused. What error codes are returned from the BE for the different scenarios? Maybe that might help my understanding.

Reply:
sure @vbahinwillit 
New codes:
10000 - Failure to fetch from VSP (Generic Error)
10005 - Failure to fetch Booked Appointments
10006 - Failure to fetch Requests

So in this case, if we get an error code of 10006, we assume there's only an error fetching requests so we only display the error message in the pending appt list view

Reply:
There's more info here. I was sent [this link] as well (https://coderepo.mobilehealth.va.gov/projects/EAS/repos/mobile-appointment-service/browse/mobile-appointment-service/src/main/java/gov/va/mobile/appointment/v2/exceptions/ErrorCode.java#67) but I don't have access so I can't speak to the contents of that page.